albums/views.py

```diff
@@ -1,3 +1,122 @@
-from django.shortcuts import render
+import json
+import math
 
-# Create your views here.
+from django.http      import JsonResponse, HttpResponse
+from django.views     import View
+from django.db.models import Q
+
+from albums.models    import (Album, Genre, Artist, ReleaseType,
+                              PlayLink, AlbumPlayLink)
+
+class ListView(View): 
+    def get(self,request):
+        
+        query       = Q()
+        filter_set  = {}
+        page        = int(request.GET.get('page',1))
+        ALBUM_COUNT = 10
+        limit       = ALBUM_COUNT*page
+        offset      = limit - ALBUM_COUNT
+
+        if request.GET.get('genre'):
+            filter_set['genre__name'] = request.GET['genre']
+        
+        if request.GET.get('year'):
+            filter_set['release_date__year'] = int(request.GET['year'])  
+
+        search_string = request.GET.get('search')
+        search_key    = request.GET.get('search_key')
+                
+        if search_string:
+            if int(search_key) == 0:
+                query &= Q(title__icontains=search_string) | Q(description__icontains=search_string)
+            elif int(search_key) == 1:
+                query &= Q(title__icontains=search_string)
+            else:
+                query &= Q(description__icontains=search_string)
+        
+        albums = Album.objects.select_related('genre','release_type').prefetch_related('artist').filter(query,**filter_set)
+        
+        if not len(albums):
```

Comment:
```suggestion
        if not albums:
```
albums가 queryset이므로, 길이로 판별하셔도 좋지만, queryset자체가 내부에 그 어떤 객체도 포함되어 있지 않으면 Falsy한 값이 됩니다 :) 따라서 len 없이 바로 not 구문을 사용하실 수 있습니다

Reply:
아! 이해됐습니다. 감사합니다. :+1: